DylibBinder/InnerX.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SwiftReflector.Inventory;
+
+namespace DylibBinder {
+	public class InnerX {
+		public InnerX ()
+		{
+		}
+
+		public Dictionary<string, List<ClassContents>> InnerXDict { get; } = new Dictionary<string, List<ClassContents>> ();
+
+		public Dictionary<string, List<ClassContents>> AddClassContentsList (params List<ClassContents>[] contents)
+		{
+			foreach (var classContentsList in contents) {
+				foreach (var classContent in classContentsList) {
+					AddItem (classContent);
+				}
+			}
+			return InnerXDict;
+		}
+
+		void AddItem (ClassContents c)
+		{
+			var nestingNames = c.Name.NestingNames.ToList ();
+			if (nestingNames.Count == 0)
+				return;
+
+			var nestingNameString = ConvertNestingNamesToString (nestingNames);
+
+			if (nestingNameString == null)
+				return;
+
+			if (nestingNames.Count == 1) {
+				AddKeyIfNotPresent (nestingNameString);
+				return;
+			}
+
+			nestingNames.RemoveAt (nestingNames.Count - 1);
+			var parentNestingNameString = ConvertNestingNamesToString (nestingNames);
+			AddKeyIfNotPresent (parentNestingNameString);
+
+			var value = InnerXDict [parentNestingNameString];
+			value.Add (c);
+			InnerXDict [parentNestingNameString] = value;
+		}
+
+		void AddKeyIfNotPresent (string key)
+		{
+			if (InnerXDict.ContainsKey (key))
+				return;
+			InnerXDict.Add (key, new List<ClassContents> ());
+		}
+
+		string ConvertNestingNamesToString (List<SwiftReflector.SwiftName> nestingNames)
```

Comment:
This is not necessary. Since `SwiftClassName` has the method `ToFullyQualifiedName`

Reply:
Thanks for pointing that out! I'm using this here and now more in other places as well!